File: js/pigLatin.js
exports.translate = function (word) {
  const vowels = ["a", "e", "i", "o", "u"];
  const word_sep = word.split(" ");
  let res_str = [];

  for (let i = 0; i < word_sep.length; i++) {
    let index = [...word_sep[i].toLowerCase()].findIndex((str) =>
      vowels.includes(str)
    );
    if (word_sep[i][index] == "u") {
      res_str.push(
        word_sep[i].slice(index + 1) + word_sep[i].slice(0, index + 1) + "ay"
      );
    } else if (index) {
      res_str.push(
        word_sep[i].slice(index) + word_sep[i].slice(0, index) + "ay"
      );
    } else {
      res_str.push(word_sep[i] + "ay");
    }
  }
  return res_str.join(" ");
};

// console.log(translate("apple"));
// console.log(translate("apple university"));
// console.log(translate("banana"));
// console.log(translate("cherry"));
// console.log(translate("eat pie"));
// console.log(translate("three"));
// console.log(translate("school"));
// console.log(translate("quiet"));
// console.log(translate("square"));
// console.log(translate("the quick brown fox"));

File: python/pig_latin.py

import re
def translate(word_or_phrase):
  vowels = ['a','e','i','o','u']
  words = word_or_phrase.split()

  for i, word in enumerate(words):

    if word[0] in vowels:
        words[i] = words[i]+ "ay"
    else:
        has_vowel = False
        
        for j, letter in enumerate(word):
            if letter in vowels:
                  if letter == 'u':
                    words[i] = word[j + 1:] + word[:j + 1]
                    has_vowel == True
                  else :
                    words[i] = word[j:] + word[:j] + "ay"
                    has_vowel = True
                    break
        if(has_vowel == False):
            words[i] = words[i]+ "ay"

  pig_latin = ' '.join(words)
  return pig_latin
    

    
    


# print(translate("apple university"))
# print(translate("square"))